reVX/least_cost_xmission/transmission_layer_creator_cli.py

```diff
@@ -0,0 +1,256 @@
+
+CONFIG_ACTIONS = ['friction_layers', 'barrier_layers', 'wet_costs',
+                  'dry_costs', 'combine_costs', 'merge_friction_and_barriers']
+
+
+@click.group()
+@click.version_option(version=__version__)
+@click.option('--verbose', '-v', is_flag=True, default=False,
+              help='Flag to turn on debug logging. Default is not verbose.')
+def main(verbose):
+    """
+    Transmission LCP routing cost, friction, and barrier layer creator.
+    """
+    log_modules = [__name__, 'reVX', 'reV', 'rex']
+    init_mult('trans-creator', None, modules=log_modules, verbose=verbose)
+
+
+@main.command
+@click.option('-c', '--config', 'config_fpath', type=click.Path(exists=True),
+              required=True, help='Configuration JSON.')
+@click.option('--ignore-unknown-keys', '-i', is_flag=True, default=False,
+              help='Silently ignore unknown keys in config file.')
+def from_config(config_fpath: str, ignore_unknown_keys: bool):  # noqa: C901
+    """
+    Create costs, barriers, and frictions from a config file.
+    """
+    # By default, throw error on unknown keys in config file
+    class ForbidExtraConfig(LayerCreationConfig, extra='forbid'):
+        """ Throw error if unknown keys are found in JSON """
+
+    ConfigClass = (LayerCreationConfig
+                   if ignore_unknown_keys else ForbidExtraConfig)
+
+    with open(config_fpath, 'r') as inf:
+        raw_json = inf.read()
+    try:
+        config = ConfigClass.model_validate_json(raw_json)
+    except ValidationError as e:
+        logger.error(f'Error loading config file {config_fpath}:\n{e}')
+        sys.exit(1)
+
+    if not any(map(lambda key: config.model_dump()[key] is not None,
+                   CONFIG_ACTIONS)):
+        logger.error(f'At least one of {CONFIG_ACTIONS} must be in the '
+                     'config file')
+        sys.exit(1)
+
+    # Done with guard clauses
+    save_tiff = config.save_tiff
+    io_handler = TransLayerIoHandler(str(config.template_raster_fpath),
+                                     layer_dir=config.layer_dir)
+    io_handler.set_h5_file(str(config.h5_fpath))
+
+    masks = Masks(io_handler, masks_dir=config.masks_dir)
+    masks.load_masks()
+
+    # Perform actions in config
+    if config.barrier_layers is not None:
+        fbb = FrictionBarrierBuilder('barrier', io_handler, masks)
+        fbb.build_layer(config.barrier_layers)
+
+    if config.friction_layers is not None:
+        fbb = FrictionBarrierBuilder('friction', io_handler, masks)
+        fbb.build_layer(config.friction_layers)
+
+    if config.wet_costs is not None:
+        wc = config.wet_costs
+        wcc = WetCostCreator(io_handler)
+        if wc.wet_costs_tiff is None:
+            wcc.build_wet_costs(str(wc.bathy_tiff), wc.bins)
+        else:
+            wcc.build_wet_costs(
+                str(wc.bathy_tiff), wc.bins, str(wc.wet_costs_tiff)
+            )
+
+    if config.dry_costs is not None:
+        # TODO - implement this
```

Comment:
Is this implementation planned under this effort?

Reply:
It probably should be. I was hoping to get this PR complete before moving onto integrating the dry cost creator into the new layers creator.